packages/helper-backend/src/deepchat-proxy/deepchat-proxy.controller.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Body,
+  Controller,
+  Param,
+  Post,
+  Get,
+  Put,
+  Delete,
+  NotFoundException,
+  Req,
+} from '@nestjs/common';
+import { DeepchatProxyService } from './deepchat-proxy.service';
+import { DeepchatProxy } from './deepchat-proxy.schema';
+import { ProxyCompletion } from './dtos/proxy-completion.dto';
+
+@Controller('deepchat-proxy')
+export class DeepchatProxyController {
+  constructor(private readonly deepchatProxyService: DeepchatProxyService) {}
+
+  @Get('/:id')
+  async get(@Param('id') id: string): Promise<DeepchatProxy> {
+    const mapping = await this.deepchatProxyService.get(id);
+    if (!mapping) {
+      throw new NotFoundException(`No model with ${id} found`);
+    }
+    return mapping;
+  }
+
+  @Get()
+  async getAll(): Promise<DeepchatProxy[]> {
+    return this.deepchatProxyService.getAll();
+  }
+
+  @Post()
+  async create(@Body() mapping: DeepchatProxy): Promise<DeepchatProxy> {
+    //return this.deepchatProxyService.create(mapping);
+    return {} as DeepchatProxy;
+  }
+
+  @Delete('/:model')
+  async delete(@Param('model') model: string): Promise<void> {
+    await this.deepchatProxyService.delete(model);
+  }
+
+  @Post('proxy/:id')
+  async proxyRequest(@Body() request: ProxyCompletion, @Param('id') id: string): Promise<any> {
+    
+    const modelData = await this.deepchatProxyService.get(id);
```

Comment:
NIT: I'd have this logic moved into the service or using a [validation pipe](https://docs.nestjs.com/pipes#custom-pipes)